libs/martech/martech.js

```diff
@@ -97,49 +98,89 @@ const getDtmLib = (env) => ({
       : env.consumer?.marTechUrl || 'https://assets.adobedtm.com/d4d114c60e50/a0e989131fd5/launch-a27b33fc2dc0-development.min.js',
 });
 
-export default async function init({ persEnabled = false, persManifests }) {
+const setupEntitlementCallback = () => {
+  const setEntitlements = async (destinations) => {
+    const { default: parseEntitlements } = await import('../features/personalization/entitlements.js');
+    return parseEntitlements(destinations);
+  };
+
+  const getEntitlements = () => new Promise((resolve) => {
+    const handleEntitlements = (e) => {
+      if (e.detail?.result?.destinations?.length) {
+        resolve(setEntitlements(e.detail.result.destinations));
+      } else {
+        resolve([]);
+      }
+    };
+    window.addEventListener(ALLOY_SEND_EVENT, handleEntitlements, { once: true });
+  });
+
+  const { miloLibs, codeRoot } = getConfig();
+  loadLink(
+    `${miloLibs || codeRoot}/features/personalization/entitlements.js`,
+    { as: 'script', rel: 'modulepreload' },
+  );
+
+  window.milo ||= {};
+  window.milo.entitlements = getEntitlements();
```

Comment:
Would it make more sense to export this method and have consuming projects import it instead of using global objects? Using this system will open the door for making other properties available and my concern is that at some point a fallback value will not be provided and personalisation / usage might get affected.

Reply:
I considered that, but I didn't want consumers to have to import the whole martech file just to get the entitlement values.  I could possibly stuff it into utils.js config which would be better, but it just seemed simpler to use `milo.entitlements`.  Definitely open to discussion though.

> my concern is that at some point a fallback value will not be provided and personalisation / usage might get affected

Yes that's possible, but I'd argue that's a dev bug if that happens.

Reply:
This pattern has been used by FEDS to expose multiple utilities, data attributes and we ended up with race conditions because there was no order in which scripts are loaded. This might not be the case with Milo, `window.milo` should always be available before consumers use it, but if `martech=off` is used, executing `window.milo.entitlements.then(console.log)` will throw an error.
I don't have a strong opinion on this, it's just based on previous experience with a similar pattern.

Reply:
After offline conversation with @narcis-radu I've moved the entitlements into the config object as a function.
If entitlements are needed, you can `await config.entitlements()`